### PR TITLE
Include opengraph/twitter metadata on AMP Story posts

### PR DESCRIPTION
**Pull request by westonruter**
See #9730.

#### Changes proposed in this Pull Request:

* In https://github.com/ampproject/amp-wp/issues/2968 a new `amp_story_head` action was introduced for plugins to output additional page metadata on AMP Story posts. This action is functionally the same as `wp_head` and is analogous to the core `embed_head` action. This PR adds outputting of the `jetpack_og_tags()` on the `amp_story` post type which has a template which does the `amp_story_head`.
* Additionally, on any AMP response this PR suppresses the HTML comments added before/after the metadata tags since the AMP plugin will re-order the nodes in the `head` for optimal performance, thus the comments would no longer appear before/after the tags.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Existing part of Jetpack: opengraph.

#### Testing instructions:

1. Do a build of the AMP plugin on the `develop` branch.
2. Activate the AMP plugin and enable the Stories experience
3. Enable the sharing module in Jetpack.
4. Try publishing a Story.
5. Verify that the expected `meta` tags are present:

```diff
7a8,23
> 	<meta property="og:type" content="article">
> 	<meta property="og:title" content="Bison">
> 	<meta property="og:url" content="https://wordpressdev.lndo.site/stories/bison-2/">
> 	<meta property="og:description" content="Bison">
> 	<meta property="article:published_time" content="2019-08-28T00:00:11+00:00">
> 	<meta property="article:modified_time" content="2019-09-03T18:01:03+00:00">
> 	<meta property="og:site_name" content="WordPressDevs">
> 	<meta property="og:image" content="https://wordpressdev.lndo.site/content/uploads/2019/08/American_bison_k5680-1-4.jpg">
> 	<meta property="og:image:width" content="1200">
> 	<meta property="og:image:height" content="783">
> 	<meta property="og:image:alt" content="Bizon!!">
> 	<meta property="og:locale" content="en_US">
> 	<meta name="twitter:text:title" content="Bison">
> 	<meta name="twitter:image" content="https://wordpressdev.lndo.site/content/uploads/2019/08/American_bison_k5680-1-4.jpg?w=640">
> 	<meta name="twitter:image:alt" content="Bizon!!">
> 	<meta name="twitter:card" content="summary_large_image">
```

#### Proposed changelog entry for your changes:

* Include opengraph metadata to AMP Story posts.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -657,6 +657,7 @@ private function __construct() {
 		 * They check for external files or plugins, so they need to run as late as possible.
 		 */
 		add_action( 'wp_head', array( $this, 'check_open_graph' ),       1 );
+		add_action( 'amp_story_head', array( $this, 'check_open_graph' ),1 );
```
Comment (jeherve) [inline class.jetpack.php]:
```suggestion
		add_action( 'amp_story_head', array( $this, 'check_open_graph' ), 1 );
```

**Review by jeherve — CHANGES_REQUESTED** [review]

**Comment by jeherve** [comment]
This should be okay short of a minor nitpick.

File: class.jetpack.php (hunk 1 above)
Comment (westonruter) [reply class.jetpack.php]: Oops.

But I just realized that if this is applied, then the result would not be aligned:

```php
		add_action( 'wp_head', array( $this, 'check_open_graph' ),       1 );
		add_action( 'amp_story_head', array( $this, 'check_open_graph' ), 1 );
		add_action( 'plugins_loaded', array( $this, 'check_twitter_tags' ),     999 );
		add_action( 'plugins_loaded', array( $this, 'check_rest_api_compat' ), 1000 );
```

So I just went ahead and removed the alignment: 1ee745d76

**Review by jeherve — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
Had to rebase. Still works well. Merging.

**Pull request merged**